### PR TITLE
feat: add lowerCase and upperCase

### PR DESCRIPTION
**Pull request by jly36963**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -416,6 +416,19 @@ const result = trimStart(str)
 
 ## Strongly-typed alternatives to common loosely-typed functions
 
+### lowerCase
+
+This function converts a string to `lower case` at both runtime and type levels.
+_NOTE: this function will split by words and join them with `" "`._
```
Comment (jly36963) [inline README.md:422]: Suggestion for a better warning?
I want it to be crystal clear that `toUpperCase` and `upperCase` are significantly different
(Same with the lower counterparts)

File: src/casing.ts [hunk 2]
```diff
@@ -80,6 +88,36 @@ function toCamelCase<T extends string>(str: T): CamelCase<T> {
   return uncapitalize(toPascalCase(str))
 }
 
+/**
+ * Transforms a string to lowercase, with words delimited by a space.
+ */
+type LowerCase<T extends string> = Join<LowerCaseAll<Words<T>>, ' '>
```
Comment (gustavoguichard) [inline src/casing.ts:94]:
```suggestion
type LowerCase<T extends string> = Lowercase<DelimiterCase<T, ' '>>

function lowerCase<T extends string>(str: T): LowerCase<T> {
  return toLowerCase(delimiterCase(str, ' '))
}
```

Comment (gustavoguichard) [reply src/casing.ts:94]: No need for `LowerCaseAll`

File: README.md [hunk 3]
```diff
@@ -416,6 +416,19 @@ const result = trimStart(str)
...
+
+```ts
+import { lowerCase } from 'string-ts'
+
+const str = 'HELLO WORLD'
```
Comment (gustavoguichard) [inline README.md:427]: Maybe make it clearer
```suggestion
const str = 'HELLO [WORLD]...'
```

File: src/internals.ts [hunk 4]
```diff
@@ -59,6 +54,35 @@ type PascalCaseAll<T extends string[]> = T extends [
...
+ */
+type UpperCaseAll<T extends string[]> = T extends [
+  infer head extends string,
+  ...infer rest extends string[],
+]
+  ? [Uppercase<head>, ...UpperCaseAll<rest>]
+  : T
+function upperCaseAll<T extends string[]>(words: T) {
+  return words.map(toUpperCase) as UpperCaseAll<T>
+}
```
Comment (gustavoguichard) [inline src/internals.ts:85]:
```suggestion
```

**Review by gustavoguichard — APPROVED** [review]
LGTM

**Pull request merged**